pkg/cluster/cluster.go

```diff
@@ -8,13 +8,14 @@ type ConnectArguments struct {
 	OfflineAccessToken      string
 	ForceCreationWithoutAsk bool
 	IgnoreContext           bool
-	SelectedKafka           string
 	Namespace               string
+	SelectedService         string
+	SelectedServiceID       string
 }
 
 // Cluster defines methods used to interact with a cluster
 type Cluster interface {
-	Connect(ctx context.Context, opts *ConnectArguments) error
+	Connect(ctx context.Context, connectOpts *ConnectArguments, c CustomConnection, opts Options) error
```

Comment:
I have not dived into the code in my editor to see what's happening here, but why is there a need for so many arguments? And then two sets of options?

We should be striving for 2 arguments maximum. There are going to be exceptions to that, but does this have to have this many?

Reply:
Context is not needed - should be in options. We have two sets of arguments because with Rama we try to keep structures with responsibilities separate. Input arguments and kubernetes client that can be reused across different commands etc.